// ng-persian-datepicker/esm2020/lib/ng-persian-datepicker.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgPersianDatepickerComponent } from './ng-persian-datepicker.component';
import { MonthPipe } from './pipe/month.pipe';
import * as i0 from "@angular/core";
export class NgPersianDatepickerModule {
}
NgPersianDatepickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.3", ngImport: i0, type: NgPersianDatepickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgPersianDatepickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.3", ngImport: i0, type: NgPersianDatepickerModule, declarations: [NgPersianDatepickerComponent,
        MonthPipe], imports: [CommonModule], exports: [NgPersianDatepickerComponent] });
NgPersianDatepickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.3", ngImport: i0, type: NgPersianDatepickerModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.3", ngImport: i0, type: NgPersianDatepickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgPersianDatepickerComponent,
                        MonthPipe
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        NgPersianDatepickerComponent
                    ]
                }]
        }] });
